samples/go/csv/read.go

```diff
@@ -131,22 +161,31 @@ func ReadToMap(r *csv.Reader, structName string, headersRaw []string, pkIdx int,
 			panic(err)
 		}
 
-		var pk types.Value
 		fields := make(types.ValueSlice, len(headersRaw))
 		for i, v := range row {
 			if i < len(headersRaw) {
 				fieldOrigIndex := fieldOrder[i]
 				fields[fieldOrigIndex], err = StringToValue(v, kindMap[fieldOrigIndex])
-				if i == pkIdx {
-					pk = fields[fieldOrigIndex]
-				}
 				if err != nil {
 					d.Chk.Fail(fmt.Sprintf("Error parsing value for column '%s': %s", headersRaw[i], err))
 				}
 			}
 		}
-		kvChan <- pk
-		kvChan <- types.NewStructWithType(t, fields)
+
+		// work backward through the pks so that outermost one is the first/primary one
+		// e.g. map<pk1, map<pk2, map<pk3, fields>>>
+		var v types.Value
+		v = types.NewStructWithType(t, fields)
+		for i := len(pkIndices); i > 1; i-- {
```

Comment:
I'm not sure if this gives us the results we're expecting. Let's talk offline and see if we're on the same page.